satpy/tests/reader_tests/test_fci_l1c_nc.py

```diff
@@ -609,14 +695,28 @@ def _get_assert_attrs(self,res,ch,attrs_dict):
         for key,item in attrs_dict.items():
             assert res[ch].attrs[key] == item
 
-    def _get_assert_load(self,res,ch,grid_type,dict_arg):
+    def _get_assert_erased_attrs(self,res,ch):
+        """Test that the attributes listed have been erased."""
+        LIST_ATTRIBUTES = ["add_offset","warm_add_offset","scale_factor",
+                           "warm_scale_factor","valid_range"]
+        for atr in LIST_ATTRIBUTES:
+            assert atr not in res[ch].attrs
+
+    def _get_assert_load(self,res,ch,grid_type,dict_arg,filenames):
         """Test the value for differents channels."""
         assert res[ch].shape == (GRID_TYPE_INFO_FOR_TEST_CONTENT[grid_type]["nrows"],
                                      GRID_TYPE_INFO_FOR_TEST_CONTENT[grid_type]["ncols"])
         assert res[ch].dtype == dict_arg["dtype"]
         self._get_assert_attrs(res,ch,dict_arg["attrs_dict"])
+        if dict_arg["attrs_dict"]["calibration"] in ["radiance","brightness_temperature","reflectance"]:
+           self._get_assert_erased_attrs(res,ch)
         if dict_arg["attrs_dict"]["calibration"] == "reflectance":
-            numpy.testing.assert_array_almost_equal(res[ch], 100 * 15 * 1 * np.pi / 50)
+            if "IQTI" in filenames:
+                numpy.testing.assert_array_almost_equal(res[ch],
+                                                93.6462)
+            else :
+                numpy.testing.assert_array_almost_equal(res[ch],
+                                                100 * 15 * 1 * np.pi / 50)
```

Comment:
CodeScene is complaining about code complexity here. Maybe you could move these ifs to separate methods (e.g. one for checking the reflectance case and one for checking the other cases).